src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeWriter.cs

```diff
@@ -651,5 +656,46 @@ public void Dispose()
             {
             }
         }
+
+        public struct StringLiteralWriter : IDisposable
```

Comment:
Whenever you cal BuildStringLiteral you'd write out the verbatim string literal start and whenever you disposed it you'd write out the string end. Either way, we can refactor later. 

Don't want you to burn too many cycles on it 😉